posthog/middleware.py

```diff
@@ -645,7 +645,48 @@ def __init__(self, get_response):
     def __call__(self, request: HttpRequest):
         # NOTE: This should be covered by the post_login signal, but we add it here as a fallback
         get_or_set_session_cookie_created_at(request=request)
-        return self.get_response(request)
+
+        if request.user.is_authenticated:
+            # Get session creation time
+            session_created_at = request.session.get(settings.SESSION_COOKIE_CREATED_AT_KEY)
+            if session_created_at:
+                # Get the shortest session timeout from all organizations the user belongs to
+                shortest_age = None
+                shortest_idle_timeout = None
+                for org in request.user.organizations.all():
```

Comment:
I thought your impl was going to be based on the current_organization which I would have gone for. 

This feels a little odd in that one org affects all the others... but maybe thats fine?

Bigger thing is that this will be adding a second query loading _alot_ of data. Can we perhaps do this on the user model somehow and optimize to only load the minimum values, ideally via a join or something?

Reply:
I wonder if I can just check the current org because when switching projects it'll update the org and check the session based on the org. 

I'll give that a try.